src/zope/i18n/tests/test_plurals.py

```diff
@@ -0,0 +1,109 @@
+                         'Es gibt eine Datei.')
+        self.assertEqual(catalog.getPluralMessage(
+                         'There is one file.', 'There are %d files.', 3),
+                         'Es gibt 3 Dateien.')
+
+        self.assertEqual(catalog.getPluralMessage(
+                         'There is one file.', 'There are %d files.', 0),
+                         'Es gibt 0 Dateien.')
+        
+        # Unknown id
+        self.assertRaises(KeyError, catalog.getPluralMessage, 
+                          'There are %d files.', 'bar', 6)
+
+        # Query without default values
+        self.assertEqual(catalog.queryPluralMessage(
+                         'There is one file.', 'There are %d files.', 1),
+                         'Es gibt eine Datei.')
+        self.assertEqual(catalog.queryPluralMessage(
+                         'There is one file.', 'There are %d files.', 3),
+                         'Es gibt 3 Dateien.')
+
+        # Query with default values
+        self.assertEqual(catalog.queryPluralMessage(
+                         'There are %d files.', 'There is one file.', 1,
+                         'Es gibt 1 Datei.', 'Es gibt %d Dateien !', ),
+                         'Es gibt 1 Datei.')
+        self.assertEqual(catalog.queryPluralMessage(
+                         'There are %d files.', 'There is one file.', 3,
+                         'Es gibt 1 Datei.', 'Es gibt %d Dateien !', ),
+                         'Es gibt 3 Dateien !')
+
+    def test_PolishPlurals(self):
+        """Polish has a complex rule for plurals. It makes for a good
+        test subject.
+        """
+        catalog = self._getMessageCatalog('pl')
+        self.assertEqual(catalog.language, 'pl')
+
+        self.assertEqual(catalog.getPluralMessage(
+                         'There is one file.', 'There are %d files.', 0),
+                         "Istnieją 0 plików.")
+
+        self.assertEqual(catalog.getPluralMessage(
+                         'There is one file.', 'There are %d files.', 1),
+                         "Istnieje 1 plik.")
+
+        self.assertEqual(catalog.getPluralMessage(
+                         'There is one file.', 'There are %d files.', 3),
+                         "Istnieją 3 pliki.")
+
+        self.assertEqual(catalog.getPluralMessage(
+                         'There is one file.', 'There are %d files.', 17),
+                         "Istnieją 17 plików.")
+
+        self.assertEqual(catalog.getPluralMessage(
+                         'There is one file.', 'There are %d files.', 23),
+                         "Istnieją 23 pliki.")
+
+        self.assertEqual(catalog.getPluralMessage(
+                         'There is one file.', 'There are %d files.', 28),
+                         "Istnieją 28 plików.")
+
+
+def test_suite():
+    return unittest.TestSuite((
+        unittest.makeSuite(TestPlurals),
+    ))
+
+if __name__ == '__main__':
+    unittest.main(defaultTest='test_suite')
```

Comment:
This is not necessary either; we've been removing these as we modernize other zope packages.